### PR TITLE
Don’t erase payment details w/ ‘update_checkout’

### PR DESCRIPTION
**Pull request by WPprodigy**
Taking a stab at https://github.com/woocommerce/woocommerce/issues/12713

Is it secure to have this object in the ajax call? It's not coming through in the response or anything, just not sure ¯\\\_(ツ)\_/¯ 

So tl;dr - Don't merge this without a hefty security review :)

Here's a demo video of this in action: http://cld.wthms.co/9YfZ/4GbeHVsu

### CODE REVIEW

**Comment by claudiosanches** [comment]
We need to check if is PCI compliance, not sure, but maybe there is something about it.

**Comment by mikejolley** [comment]
@claudiosanches what is the concern if the details are never sent to the server? If they are client side, it's no more a risk than scripts scraping the fields directly vs the data stored.

File: assets/js/frontend/checkout.js [hunk 1]
```diff
@@ -303,6 +317,21 @@ jQuery( function( $ ) {
...
+					// Fill in the payment details if possible
+					if ( ! $.isEmptyObject( paymentDetails ) ) {
+						$( '.payment_box input' ).each( function() {
+							var ID = $( this ).attr( 'id' );
+
+							if ( ID ) {
+								if ( 'checkbox' === $( this ).attr( 'type' ) && paymentDetails[ ID ] ) {
+									$( this ).prop( 'checked', true );
+								} else {
+									$( this ).val( paymentDetails[ ID ] );
```
Comment (mikejolley) [inline assets/js/frontend/checkout.js]: Trigger change() after these so any logic that hooks into change is fired. E.g. token generation.

Can you test with Stripe?

**Comment by mikejolley** [comment]
Added 1 comment which needs some attention @WPprodigy 

Security wise it looks fine as it's client side only - I checked with security team too.

**Comment by WPprodigy** [comment]
Added .change() and fixed an issue w/ radio buttons not working.

**Pull request merged**